client/app/components/TypePicker.jsx

```diff
@@ -0,0 +1,85 @@
+import Card from 'antd/lib/card';
+import Input from 'antd/lib/input';
+import List from 'antd/lib/list';
+import { includes, isEmpty } from 'lodash';
+import PropTypes from 'prop-types';
+import React from 'react';
+import { Type } from './proptypes';
+import EmptyState from '@/components/items-list/components/EmptyState';
+
+const { Search } = Input;
+const { Meta } = Card;
+
+export default class TypePicker extends React.Component {
+  static propTypes = {
+    types: PropTypes.arrayOf(Type),
+  };
+
+  static defaultProps = {
+    types: [],
+  };
+
+  constructor(props) {
+    super(props);
+    this.state = { searchText: '' };
+  }
+
+  // eslint-disable-next-line class-methods-use-this
+  renderListItem(item) {
+    const titleStyle = {
+      fontSize: '13px',
+      maxHeight: '50px',
+      whiteSpace: 'normal',
+      textOverflow: 'ellipsis',
+    };
+
+    return (
+      <List.Item>
+        <Card
+          bodyStyle={{ height: '80px', padding: '15px' }}
+          cover={(
+            <div className="m-t-10">
+              <img
+                alt={item.name}
+                style={{ margin: 'auto', width: '64px', height: '64px' }}
+                src={item.imgSrc}
+              />
+            </div>
+          )}
+          onClick={item.onClick}
+          hoverable
+        >
+          <Meta title={(<p style={titleStyle}>{item.name}</p>)} />
+        </Card>
+      </List.Item>
+    );
+  }
+
+  render() {
+    const { types } = this.props;
+    const { searchText } = this.state;
+
+    const filteredTypes = types.filter(type => isEmpty(searchText) ||
+      includes(type.name.toLowerCase(), searchText.toLowerCase()));
+
+    return (
+      <div className="text-center" data-test="TypePicker">
+        <Search
+          className="m-b-20"
+          placeholder="Search..."
+          onChange={e => this.setState({ searchText: e.target.value })}
+          style={{ maxWidth: 300 }}
+          autoFocus
+        />
+        {isEmpty(filteredTypes) ? (<EmptyState />) : (
```

Comment:
By default `EmptyState` and `LoadingState` have an outline ('.tiled' class set) - check if it works as expected, maybe you'll need to override this.